ci/run-ci.py

```diff
@@ -0,0 +1,475 @@
+    # Gcloud Build artifacts
+    artifacts_file = os.path.join(SOURCE_DIR, config['name'], 'tags.json')
+
+    with open(artifacts_file, 'w') as file:
+        tags = {'builds': []}
+        for image in config['images']:
+            tags['builds'].append({
+                'imageName': f'substrafoundation/{image}',
+                'tag': f'gcr.io/{CLUSTER_PROJECT}/{image}:{config["commit"]}'
+            })
+
+        json.dump(tags, file)
+
+    return artifacts_file
+
+
+def patch_skaffold_file(config):
+
+    skaffold_file = os.path.join(SOURCE_DIR, config['name'], 'skaffold.yaml')
+
+    # Patch skaffold file
+    with open(skaffold_file, 'r') as file:
+        filedata = file.read()
+
+    # Set statusCheckDeadlineSeconds to 300s instead of 120 default
+    filedata = filedata.replace(
+        'deploy:',
+        'deploy:\n  statusCheckDeadlineSeconds: 300'
+    )
+
+    filedata = filedata.replace(
+        'chartPath: charts/',
+        f'chartPath: {os.path.join(SOURCE_DIR, config["name"],"charts/")}'
+    )
+
+    if config['name'] == 'substra-backend':
+        # Default Domain
+        filedata = filedata.replace(
+            'defaultDomain: http://substra-backend.node-1.com',
+            'defaultDomain: http://backend-org-1-substra-backend-server.org-1.svc.cluster.local:8000'
+        )
+        filedata = filedata.replace(
+            'defaultDomain: http://substra-backend.node-2.com',
+            'defaultDomain: http://backend-org-2-substra-backend-server.org-2.svc.cluster.local:8000'
+        )
+
+    with open(skaffold_file, 'w') as file:
+        file.write(filedata)
+
+    return skaffold_file
+
+
+def run_tests():
+    print('# Wait for the substra-tests pod to be ready')
+    substra_tests_pod = call_output(
+        f'kubectl --context {KUBE_CONTEXT} get pods -n substra-tests | grep substra-tests'
+    ).split(' ')[0]
+
+    try:
+        call(f'kubectl --context {KUBE_CONTEXT} wait pod/{substra_tests_pod} '
+             f'-n substra-tests --for=condition=ready --timeout=590s')
+    except Exception:
+        print('ERROR: Timeout while waiting for the substra-tests pod. '
+              'This means the `substra-backend-server` pods never reached the "ready" state.')
+
+    print('\n# Run tests')
+
+    try:
+        call(f'kubectl --context {KUBE_CONTEXT} exec {substra_tests_pod} -n substra-tests -- make test')
+        return True
+    except subprocess.CalledProcessError as e:
+        print('FATAL: `make test` completed with a non-zero exit code. Did some test(s) fail?')
+        return False
+
+
+def main():
+    is_success=False
+    arg_parse()
+
+    try:
+        gcloud_login()
+        create_cluster_async()
+        configs = clone_repos()
+        build_images(configs)
+        wait_for_cluster()
+        get_kube_context()
+        setup_tiller()
+        deploy_all(configs)
+        is_success = run_tests()
+
+    except Exception as e:
+        print(f'FATAL: {e}')
+        is_success=False
```

Comment:
Why printing the exception?

It would be simpler to not catch it: the traceback will be printed in stdout and the result code will be 1.

Reply:
I added a `print` here is so that the exception is displayed _before_ the `finally` block is executed, not after. If it's displayed after, the logs can be hard to read. I also suspect that if we abort the script during `finally` (Ctrl+C), the exception might not be displayed at all?

Reply:
Sounds good to me.